main_test.go

```diff
@@ -669,16 +669,22 @@ var setFieldsTestCases = []struct {
 		},
 	},
 	{
-		Name:        "source testOneof.k.a.testNestedNestedOneOf.g empty",
-		Destination: &testdata.Test{},
-		Source: &testdata.Test{
+		Name: "source testOneof.k.a.testNestedNestedOneOf.g empty",
```

Comment:
The updated test case would panic at runtime before the fix. This is caused by the fact that having a `nil` source message is supported with semantics of 'unset leaf'.